crates/core/src/client/mod.rs

```diff
@@ -499,12 +488,68 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
     }
 
     /// Returns the estimated gas for a transaction
-    async fn estimate_gas(
-        &self,
-        _call_request: CallRequest,
-        _block_number: Option<BlockId>,
-    ) -> Result<U256, EthApiError<P::Error>> {
-        todo!();
+    async fn estimate_gas(&self, request: CallRequest, block_id: BlockId) -> Result<U256, EthApiError<P::Error>> {
+        let chain_id: u64 = request
+            .chain_id
+            .ok_or_else(|| EthApiError::MissingParameterError("chain_id for estimate_gas".into()))?
+            .low_u64();
+
+        let from = request.from.ok_or_else(|| EthApiError::MissingParameterError("from for estimate_gas".into()))?;
+        let nonce: Felt252Wrapper = self.nonce(from, block_id).await?.try_into()?;
+
+        let gas_limit: Felt252Wrapper = request.gas.unwrap_or_else(|| U256::from(1_000_000)).try_into()?;
+        let max_fee_per_gas: Felt252Wrapper =
+            request.max_fee_per_gas.unwrap_or_else(|| U256::from(BASE_FEE_PER_GAS)).try_into()?;
+        let max_priority_fee_per_gas: Felt252Wrapper =
+            request.max_priority_fee_per_gas.unwrap_or_else(|| U256::from(MAX_PRIORITY_FEE_PER_GAS)).try_into()?;
+
+        let to = request.to.map_or(TransactionKind::Create, TransactionKind::Call);
+
+        let value: Felt252Wrapper = request.value.unwrap_or_else(|| U256::from(0)).try_into()?;
+
+        let data = request.data.unwrap_or_default();
```

Comment:
Have we checked what the default is for type of data?
I think bytes used to have tricky default

Reply:
Default for Bytes is a ptr to an empty slice of [u8], I think default is what we need.